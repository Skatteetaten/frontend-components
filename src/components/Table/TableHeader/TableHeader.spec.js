import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TableHeader } from '.';

const columns = [
  {
    fieldName: 'month',
  },
  {
    name: 'Beløp',
    fieldName: 'amount',
    alignment: 'right',
    sortable: true,
  },
  {
    name: 'Dekningsgrad',
    fieldName: 'coverage',
    alignment: 'right',
  },
  {
    name: 'Avkastning',
    fieldName: 'revenue',
    alignment: 'right',
  },
];

describe('TableHeader komponent', () => {
  describe('Når columns består av en kolonne uten "name", en kolonne "sortable" og vanlig kolonner', () => {
    const setSortMock = jest.fn();
    const wrapper = mount(
      <table>
        <tbody>
          <tr>
            <TableHeader
              columns={columns}
              sort={{ ascending: true, columnFieldName: 'month' }}
              setSort={setSortMock}
            />
          </tr>
        </tbody>
      </table>
    );

    it('Så ser markupen riktig ut', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      // Det skal bare vises 3 th i og med at column uten 'name' vises ikke
      expect(wrapper.find('th').length).toEqual(3);
      // Det skal bare vises ett icon i og med at bare én column er sorterbar
      expect(wrapper.find('Icon').length).toEqual(1);
    });
    it('Så kalles setSort function riktig ved klikke på en sortable kolonne-header ', () => {
      wrapper.find('th').at(0).simulate('click');
      expect(setSortMock).toHaveBeenCalled();
    });
  });
});
