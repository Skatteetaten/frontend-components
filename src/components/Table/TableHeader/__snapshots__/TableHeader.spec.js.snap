// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader komponent Når columns består av en kolonne uten "name", en kolonne "sortable" og vanlig kolonner Så ser markupen riktig ut 1`] = `
<TableHeader
  columns={
    Array [
      Object {
        "fieldName": "month",
      },
      Object {
        "alignment": "right",
        "fieldName": "amount",
        "name": "Beløp",
        "sortable": true,
      },
      Object {
        "alignment": "right",
        "fieldName": "coverage",
        "name": "Dekningsgrad",
      },
      Object {
        "alignment": "right",
        "fieldName": "revenue",
        "name": "Avkastning",
      },
    ]
  }
  setSort={[MockFunction]}
  sort={
    Object {
      "ascending": true,
      "columnFieldName": "month",
    }
  }
>
  <td
    className=

        {
          border-bottom: 2px solid #1a1a1a;
        }
    key="month"
  />
  <th
    aria-label="Beløp sorterbar"
    className=

        {
          border-bottom: 2px solid #1a1a1a;
          font-size: inherit;
          font-weight: 700;
          padding-bottom: 1rem;
          padding-left: 1rem;
          padding-right: 1rem;
          padding-top: 1rem;
          position: relative;
          vertical-align: bottom;
        }
        & i {
          color: #1362ae;
          cursor: pointer;
          padding-left: 0.125rem;
          position: absolute;
        }
        & i:hover {
          color: #093e61;
        }
        @media (min-width: 1024px){& [data-icon-name="ArrowUpDown"] {
          opacity: 0;
        }
        & .noAutoHide {
          opacity: 1;
        }
        & [data-icon-name="ArrowUpDown"]:hover {
          opacity: 1;
        }
        &:focus {
          color: #1362ae;
          outline: 0;
        }
        {
          cursor: pointer;
        }
        &:hover [data-icon-name="ArrowUpDown"] {
          opacity: 1;
        }
        &:focus [data-icon-name="ArrowUpDown"] {
          opacity: 1;
        }
        {
          justify-content: flex-end;
          text-align: right;
        }
    key="amount"
    onClick={[Function]}
    onKeyDown={[Function]}
    scope="col"
    tabIndex={0}
  >
    Beløp
    <Icon
      className="noAutoHide"
      iconName="ArrowUpDown"
    >
      <StyledIconBase
        className="noAutoHide"
        iconName="ArrowUpDown"
      >
        <IconBase
          className="noAutoHide"
          iconName="ArrowUpDown"
          styles={[Function]}
          theme={
            Object {
              "disableGlobalClassNames": false,
              "effects": Object {
                "elevation16": "0 6.4px 14.4px 0 rgba(0, 0, 0, 0.132), 0 1.2px 3.6px 0 rgba(0, 0, 0, 0.108)",
                "elevation4": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
                "elevation64": "0 25.6px 57.6px 0 rgba(0, 0, 0, 0.22), 0 4.8px 14.4px 0 rgba(0, 0, 0, 0.18)",
                "elevation8": "0 3.2px 7.2px 0 rgba(0, 0, 0, 0.132), 0 0.6px 1.8px 0 rgba(0, 0, 0, 0.108)",
                "roundedCorner2": "2px",
                "roundedCorner4": "4px",
                "roundedCorner6": "6px",
              },
              "fonts": Object {
                "large": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "1.125rem",
                  "fontWeight": 300,
                },
                "medium": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "1rem",
                  "fontWeight": 400,
                },
                "mediumPlus": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "1.0625rem",
                  "fontWeight": 400,
                },
                "mega": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "4.25rem",
                  "fontWeight": 100,
                },
                "small": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "0.875rem",
                  "fontWeight": 400,
                },
                "smallPlus": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "0.9375rem",
                  "fontWeight": 400,
                },
                "superLarge": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "2.625rem",
                  "fontWeight": 100,
                },
                "tiny": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "0.625rem",
                  "fontWeight": 600,
                },
                "xLarge": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "1.375rem",
                  "fontWeight": 100,
                },
                "xLargePlus": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
                  "fontSize": "24px",
                  "fontWeight": 600,
                },
                "xSmall": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "0.75rem",
                  "fontWeight": 400,
                },
                "xxLarge": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Helvetica Neue', Helvetica , Arial, 'Liberation Sans', 'Nimbus Sans L', sans-serif",
                  "fontSize": "1.875rem",
                  "fontWeight": 100,
                },
                "xxLargePlus": Object {
                  "MozOsxFontSmoothing": "grayscale",
                  "WebkitFontSmoothing": "antialiased",
                  "fontFamily": "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif",
                  "fontSize": "32px",
                  "fontWeight": 600,
                },
              },
              "isInverted": false,
              "palette": Object {
                "accent": "#1362ae",
                "black": "#1a1a1a",
                "blackTranslucent40": "rgba(0,0,0,.4)",
                "blue": "#0078d4",
                "blueDark": "#002050",
                "blueLight": "#00bcf2",
                "blueMid": "#00188f",
                "bodyBackground": "#ffffff",
                "bodyText": "rgba(0, 0, 0, 0.95)",
                "disabledBackground": "#f4f4f4",
                "disabledText": "#808080",
                "green": "#107c10",
                "greenDark": "#004b1c",
                "greenLight": "#bad80a",
                "magenta": "#b4009e",
                "magentaDark": "#5c005c",
                "magentaLight": "#e3008c",
                "neutralDark": "rgba(21, 21, 21, 0.95)",
                "neutralLight": "#eaeaea",
                "neutralLighter": "#f4f4f4",
                "neutralLighterAlt": "#f8f8f8",
                "neutralPrimary": "rgba(0, 0, 0, 0.95)",
                "neutralPrimaryAlt": "rgba(47, 47, 47, 0.95)",
                "neutralQuaternary": "#d0d0d0",
                "neutralQuaternaryAlt": "#dadada",
                "neutralSecondary": "rgba(55, 55, 55, 0.95)",
                "neutralSecondaryAlt": "#8a8886",
                "neutralTertiary": "rgba(89, 89, 89, 0.95)",
                "neutralTertiaryAlt": "#c8c8c8",
                "orange": "#d83b01",
                "orangeLight": "#ea4300",
                "orangeLighter": "#ff8c00",
                "primaryBackground": "#ffffff",
                "primaryText": "#1a1a1a",
                "purple": "#5c2d91",
                "purpleDark": "#32145a",
                "purpleLight": "#b4a0ff",
                "red": "#e81123",
                "redDark": "#a4262c",
                "skeColor": Object {
                  "backgroundFocusColor": "#e2eaf7",
                  "backgroundHoverColor": "#e2eaf7",
                  "beige": "#f9ede2",
                  "black": "#000000",
                  "black100": "#1a1a1a",
                  "blackAlt": "#1a1a1a",
                  "blue": "#1362ae",
                  "blue10": "#e2eaf7",
                  "blue100": "#2c436b",
                  "blue30": "#b5cbee",
                  "blue5": "#eff3f9",
                  "blue50": "#89abe5",
                  "blue70": "#5a77a8",
                  "brown": "#e7b78a",
                  "brown10": "#f9ede2",
                  "brown100": "#6b492c",
                  "brown30": "#f0d2b6",
                  "brown5": "#f9f4f0",
                  "brown50": "#e7b78a",
                  "brown70": "#a9805b",
                  "burgundy": "#6f2c3f",
                  "burgundy10": "#f7e2e8",
                  "burgundy100": "#6f2c3f",
                  "burgundy30": "#f3a7b0",
                  "burgundy5": "#faf0f3",
                  "burgundy50": "#f06674",
                  "burgundy70": "#b04c5c",
                  "burgundyLight": "#b04c5c",
                  "darkBeige": "#f0d2b6",
                  "darkBlue": "#093e61",
                  "darkGrey": "#4d4d4d",
                  "error": "#a82e39",
                  "green": "#91d6ac",
                  "green10": "#e3f5ea",
                  "green100": "#2b6940",
                  "green30": "#b9e1c8",
                  "green5": "#f4faf5",
                  "green50": "#91d6ac",
                  "green70": "#5d9b73",
                  "grey": "#808080",
                  "grey10": "#e5e5e5",
                  "grey30": "#b2b2b2",
                  "grey5": "#f4f4f4",
                  "grey50": "#808080",
                  "grey70": "#4d4d4d",
                  "interactive": "#1362ae",
                  "interactiveDark": "#093e61",
                  "interactiveLight": "#cde1f9",
                  "internal": "#6f2c3f",
                  "internalLight": "#faf0f3",
                  "lightBeige": "#f9f4f0",
                  "lightBlue": "#cde1f9",
                  "lightBrown": "#f0d2b6",
                  "lightGreen": "#e3f5ea",
                  "lightGrey": "#b2b2b2",
                  "lightPink": "#f7d1d7",
                  "neutralGrey": "#f4f4f4",
                  "pink": "#df4661",
                  "statusError": "#a82e39",
                  "statusOk": "#316f2a",
                  "statusWarning": "#9F7509",
                  "transparent": "transparent",
                  "white": "#ffffff",
                  "whiteGrey": "#e5e5e5",
                },
                "teal": "#008272",
                "tealDark": "#004b50",
                "tealLight": "#00b294",
                "themeDark": "#0c3f6f",
                "themeDarkAlt": "#11599c",
                "themeDarker": "#0a365f",
                "themeLight": "#82bcf1",
                "themeLighter": "#cce3f9",
                "themeLighterAlt": "#ebf4fd",
                "themePrimary": "#1362ae",
                "themeSecondary": "#146aba",
                "themeTertiary": "#288be7",
                "white": "#ffffff",
                "whiteTranslucent40": "rgba(255,255,255,.4)",
                "yellow": "#ffb900",
                "yellowDark": "#d29200",
                "yellowLight": "#fff100",
              },
              "rtl": undefined,
              "semanticColors": Object {
                "accentButtonBackground": "#1362ae",
                "accentButtonText": "#ffffff",
                "actionLink": "rgba(0, 0, 0, 0.95)",
                "actionLinkHovered": "rgba(21, 21, 21, 0.95)",
                "blockingBackground": "#FDE7E9",
                "blockingIcon": "#FDE7E9",
                "bodyBackground": "#ffffff",
                "bodyBackgroundChecked": "#eaeaea",
                "bodyBackgroundHovered": "#f4f4f4",
                "bodyDivider": "#eaeaea",
                "bodyFrameBackground": "#ffffff",
                "bodyFrameDivider": "#eaeaea",
                "bodyStandoutBackground": "#f8f8f8",
                "bodySubtext": "rgba(55, 55, 55, 0.95)",
                "bodyText": "rgba(0, 0, 0, 0.95)",
                "bodyTextChecked": "#1a1a1a",
                "buttonBackground": "#ffffff",
                "buttonBackgroundChecked": "#c8c8c8",
                "buttonBackgroundCheckedHovered": "#eaeaea",
                "buttonBackgroundDisabled": "#f4f4f4",
                "buttonBackgroundHovered": "#f4f4f4",
                "buttonBackgroundPressed": "#eaeaea",
                "buttonBorder": "#8a8886",
                "buttonBorderDisabled": "#f4f4f4",
                "buttonText": "rgba(0, 0, 0, 0.95)",
                "buttonTextChecked": "rgba(21, 21, 21, 0.95)",
                "buttonTextCheckedHovered": "#1a1a1a",
                "buttonTextDisabled": "rgba(89, 89, 89, 0.95)",
                "buttonTextHovered": "rgba(21, 21, 21, 0.95)",
                "buttonTextPressed": "rgba(21, 21, 21, 0.95)",
                "cardShadow": "0 1.6px 3.6px 0 rgba(0, 0, 0, 0.132), 0 0.3px 0.9px 0 rgba(0, 0, 0, 0.108)",
                "cardShadowHovered": "0 0 1px rgba(89, 89, 89, 0.95)",
                "cardStandoutBackground": "#ffffff",
                "defaultStateBackground": "#f8f8f8",
                "disabledBackground": "#f4f4f4",
                "disabledBodySubtext": "#c8c8c8",
                "disabledBodyText": "rgba(89, 89, 89, 0.95)",
                "disabledBorder": "#c8c8c8",
                "disabledSubtext": "#d0d0d0",
                "disabledText": "rgba(89, 89, 89, 0.95)",
                "errorBackground": "#FDE7E9",
                "errorIcon": "#A80000",
                "errorText": "#a4262c",
                "focusBorder": "rgba(55, 55, 55, 0.95)",
                "infoBackground": "#f3f2f1",
                "infoIcon": "#605e5c",
                "inputBackground": "#ffffff",
                "inputBackgroundChecked": "#1362ae",
                "inputBackgroundCheckedHovered": "#0c3f6f",
                "inputBorder": "rgba(55, 55, 55, 0.95)",
                "inputBorderHovered": "rgba(0, 0, 0, 0.95)",
                "inputFocusBorderAlt": "#1362ae",
                "inputForegroundChecked": "#ffffff",
                "inputIcon": "#1362ae",
                "inputIconDisabled": "rgba(89, 89, 89, 0.95)",
                "inputIconHovered": "#0c3f6f",
                "inputPlaceholderBackgroundChecked": "#cce3f9",
                "inputPlaceholderText": "rgba(55, 55, 55, 0.95)",
                "inputText": "rgba(0, 0, 0, 0.95)",
                "inputTextHovered": "rgba(21, 21, 21, 0.95)",
                "link": "#1362ae",
                "linkHovered": "#0a365f",
                "listBackground": "#ffffff",
                "listHeaderBackgroundHovered": "#f4f4f4",
                "listHeaderBackgroundPressed": "#eaeaea",
                "listItemBackgroundChecked": "#eaeaea",
                "listItemBackgroundCheckedHovered": "#dadada",
                "listItemBackgroundHovered": "#f4f4f4",
                "listText": "rgba(0, 0, 0, 0.95)",
                "listTextColor": "#323130",
                "menuBackground": "#ffffff",
                "menuDivider": "#c8c8c8",
                "menuHeader": "#1362ae",
                "menuIcon": "#1362ae",
                "menuItemBackgroundChecked": "#eaeaea",
                "menuItemBackgroundHovered": "#f4f4f4",
                "menuItemBackgroundPressed": "#eaeaea",
                "menuItemText": "rgba(0, 0, 0, 0.95)",
                "menuItemTextHovered": "rgba(21, 21, 21, 0.95)",
                "messageLink": "#005A9E",
                "messageLinkHovered": "#004578",
                "messageText": "#323130",
                "primaryButtonBackground": "#1362ae",
                "primaryButtonBackgroundDisabled": "#f4f4f4",
                "primaryButtonBackgroundHovered": "#11599c",
                "primaryButtonBackgroundPressed": "#0c3f6f",
                "primaryButtonBorder": "transparent",
                "primaryButtonText": "#ffffff",
                "primaryButtonTextDisabled": "#d0d0d0",
                "primaryButtonTextHovered": "#ffffff",
                "primaryButtonTextPressed": "#ffffff",
                "severeWarningBackground": "#FED9CC",
                "severeWarningIcon": "#D83B01",
                "smallInputBorder": "rgba(55, 55, 55, 0.95)",
                "successBackground": "#DFF6DD",
                "successIcon": "#107C10",
                "successText": "#107C10",
                "variantBorder": "#eaeaea",
                "variantBorderHovered": "rgba(89, 89, 89, 0.95)",
                "warningBackground": "#FFF4CE",
                "warningHighlight": "#ffb900",
                "warningIcon": "#797775",
                "warningText": "#323130",
              },
              "spacing": Object {
                "l1": "20px",
                "l2": "32px",
                "m": "16px",
                "s1": "8px",
                "s2": "4px",
              },
            }
          }
        >
          <i
            aria-hidden={true}
            className=
                noAutoHide
                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  display: inline-block;
                  font-family: "SkeIconfont";
                  font-size: inherit;
                  font-style: normal;
                  font-weight: normal;
                  speak: none;
                  text-rendering: auto;
                  transform: translate(0, 0);
                }
            data-icon-name="ArrowUpDown"
          >
            
          </i>
        </IconBase>
      </StyledIconBase>
    </Icon>
  </th>
  <th
    className=

        {
          border-bottom: 2px solid #1a1a1a;
          font-size: inherit;
          font-weight: 700;
          padding-bottom: 1rem;
          padding-left: 1rem;
          padding-right: 1rem;
          padding-top: 1rem;
          position: relative;
          vertical-align: bottom;
        }
        & i {
          color: #1362ae;
          cursor: pointer;
          padding-left: 0.125rem;
          position: absolute;
        }
        & i:hover {
          color: #093e61;
        }
        @media (min-width: 1024px){& [data-icon-name="ArrowUpDown"] {
          opacity: 0;
        }
        & .noAutoHide {
          opacity: 1;
        }
        & [data-icon-name="ArrowUpDown"]:hover {
          opacity: 1;
        }
        &:focus {
          color: #1362ae;
          outline: 0;
        }
        {
          justify-content: flex-end;
          text-align: right;
        }
    key="coverage"
    scope="col"
  >
    Dekningsgrad
  </th>
  <th
    className=

        {
          border-bottom: 2px solid #1a1a1a;
          font-size: inherit;
          font-weight: 700;
          padding-bottom: 1rem;
          padding-left: 1rem;
          padding-right: 1rem;
          padding-top: 1rem;
          position: relative;
          vertical-align: bottom;
        }
        & i {
          color: #1362ae;
          cursor: pointer;
          padding-left: 0.125rem;
          position: absolute;
        }
        & i:hover {
          color: #093e61;
        }
        @media (min-width: 1024px){& [data-icon-name="ArrowUpDown"] {
          opacity: 0;
        }
        & .noAutoHide {
          opacity: 1;
        }
        & [data-icon-name="ArrowUpDown"]:hover {
          opacity: 1;
        }
        &:focus {
          color: #1362ae;
          outline: 0;
        }
        {
          justify-content: flex-end;
          text-align: right;
        }
    key="revenue"
    scope="col"
  >
    Avkastning
  </th>
</TableHeader>
`;
